blockchain/chain.go

```diff
@@ -602,6 +611,16 @@ func (b *BlockChain) connectBlock(node *blockNode, block *btcutil.Block,
 		curTotalTxns+numTxns, CalcPastMedianTime(node),
 	)
 
+	// If a utxoviewpoint was passed in, we'll be writing that viewpoint
+	// directly to the database on disk.  In order for the database to be
+	// consistent, we must flush the cache before writing the viewpoint.
+	if view != nil {
```

Comment:
In the current code, when a reorg happens, the utxoviewpoint will be passed in. This allows for less code changes for the reorg code as we can keep the current code that uses the utxoviewpoint directly.

However, when this happens, the cache may have unwritten new utxos or spent coins that should be deleted. In order for the code to be correct, the utxoview should be modifying the utxocache. But this results in more code being needed.

To avoid interactions with the utxoviewpoint and the utxocache, we just flush the utxocache first whenever the utxoviewpoint is passed in.